mobile_verifier/src/boosting_oracles/mod.rs

```diff
@@ -2,69 +2,177 @@ pub mod assignment;
 
 use std::collections::HashMap;
 
-use crate::geofence::GeofenceValidator;
+use crate::{
+    geofence::{Geofence, GeofenceValidator},
+    Settings,
+};
 pub use assignment::Assignment;
 use hextree::disktree::DiskTreeMap;
 
-pub trait DiskTreeLike: Send + Sync + 'static {
-    fn get(&self, cell: hextree::Cell) -> hextree::Result<Option<(hextree::Cell, &[u8])>>;
+pub fn make_hex_boost_data(
+    settings: &Settings,
+    usa_geofence: Geofence,
+) -> anyhow::Result<HexBoostData<UrbanizationData<DiskTreeMap, Geofence>, FootfallData<DiskTreeMap>>>
+{
+    let urban_disktree = DiskTreeMap::open(&settings.urbanization_data_set)?;
+    let footfall_disktree = DiskTreeMap::open(&settings.footfall_data_set)?;
+
+    let urbanization = UrbanizationData::new(urban_disktree, usa_geofence);
+    let footfall_data = FootfallData::new(footfall_disktree);
+    let hex_boost_data = HexBoostData::new(urbanization, footfall_data);
+
+    Ok(hex_boost_data)
 }
 
-impl DiskTreeLike for DiskTreeMap {
-    fn get(&self, cell: hextree::Cell) -> hextree::Result<Option<(hextree::Cell, &[u8])>> {
-        self.get(cell)
-    }
+pub enum FootfallCategory {
+    /// More than 1 point of interest
+    Poi,
+    /// (ZERO) points of interest
+    PoiNoData,
+    /// Cell not found
+    NoPoi,
 }
 
-impl DiskTreeLike for HashMap<hextree::Cell, Vec<u8>> {
-    fn get(&self, cell: hextree::Cell) -> hextree::Result<Option<(hextree::Cell, &[u8])>> {
-        Ok(self.get(&cell).map(|x| (cell, x.as_slice())))
-    }
+pub enum UrbanizationCategory {
+    Urban,
+    NotUrban,
+    OutsideUSA,
 }
 
-pub struct MockDiskTree;
+pub trait HexAssignment: Send + Sync {
+    fn urban_assignment(&self, cell: hextree::Cell) -> anyhow::Result<Assignment>;
+    fn footfall_assignment(&self, cell: hextree::Cell) -> anyhow::Result<Assignment>;
+}
```

Comment:
I don't really see the purpose of bundling the two data sets together here. I would recommend implementing `HexAssignment` for both of the data sets independently, rather than having one for both:
```suggestion
pub trait HexAssignment: Send + Sync {
    fn assignment(&self, cell: hextree::Cell) -> anyhow::Result<Assignment>;
}
```